src/bootstrap/compile.rs

```diff
@@ -1102,6 +1102,9 @@ impl Step for Assemble {
             let src_exe = exe("lld", target_compiler.host);
             let dst_exe = exe("rust-lld", target_compiler.host);
             builder.copy(&lld_install.join("bin").join(&src_exe), &libdir_bin.join(&dst_exe));
+            // FIXME handle error instead of silently ignoring
+            #[cfg(target_os = "linux")]
+            let _ = std::os::unix::fs::symlink(libdir_bin.join(&dst_exe), libdir_bin.join("ld"));
```

Comment:
The target os may not be the same as the build os, so `#[cfg(target_os = "linux")]` is wrong. I think the `#[cfg(target_os = "linux")]` should even be completely removed. It would likely give problems on windows though due to symlinks only being allowed for non-privileged users when on windows 10 with the developer mode enabled.

Reply:
oh right, target_os is what we build the compiler for, but we do this step at build time now. what's the preferred way of making symlinks?

Reply:
I am not sure. cc @Mark-Simulacrum?

Reply:
wait no, the `target_os` is for the build machine of the compiler right? according to the replacements of `https://doc.rust-lang.org/std/fs/fn.soft_link.html`, maybe a `#[cfg(unix)]` with `std::os::unix::fs::symlink` and a `#[cfg(windows)]` with `std::os::windows::fs::symlink_file`?

Reply:
`symlink_file` requires admin permission when developer mode is not enabled or when not running on windows 10.

Reply:
Maybe `ld.lld` could be a small rust program running `lld` with `-flavor gnu` as first two args?

Reply:
what about hard links?

Reply:
That could work I think, though I am not sure if tar preserves hard links. You can try. The `builder.copy()` function tries to hardlink first by the way before falling back to copying.

Reply:
looks like hard links are preserved https://unix.stackexchange.com/questions/43037/dereferencing-hard-links. actually, yeah, copying sounds like a good idea.